source/common/stats/dog_statsd.cc

```diff
@@ -0,0 +1,75 @@
+#include "common/stats/dog_statsd.h"
+
+#include <chrono>
+#include <cstdint>
+#include <string>
+
+#include "envoy/stats/stats.h"
+
+#include "fmt/format.h"
+
+namespace Envoy {
+namespace Stats {
+namespace DogStatsd {
+
+Writer::Writer(Network::Address::InstanceConstSharedPtr address) : Statsd::Writer(address) {}
+
+void Writer::writeCounter(const std::string& name, const std::string& tag_str, uint64_t increment) {
+  std::string message(fmt::format("envoy.{}:{}|c{}\n", name, increment, tag_str));
```

Comment:
At a high-level, wondering if you could comment on the tradeoff between doing this as a separate sink vs. making this a configurable option on the existing `statsd` sink? It seems the formats are fairly similar, with regular `statsd a subset. Don't mind a separate sink if it makes more sense.

Reply:
Right, it's another option to implement this. In my opinion, pros of a separate sink: dedicated sink confiuration brings future extensibility on configuration around tagging. Then, pros of exteding existing `statsd` sink: compact code base and configuration.

I think we have to think which is more user-friendly idea not from point of code-base. In my current opinion, since the original issue #2024 starts from a new sink with dedicated configuration, the new sink seems more human natural idea. What do you think?

For code-base or design points, I'm welcome to any advices and will make improvements.

Reply:
@taiki45 I think having a dedicated configuration makes sense from the user perspective. I think the main question is whether to have a shared base class for the implementation (or just one implementation with some functionality variable built in). I don't really have that strong of an opinion either way as the code is pretty self contained.

Reply:
Sure that SGTM.

Reply:
Done in 5937ac4f5602fb46bd0e03eb45ebd9c20deeecec.